src/VisualStudio/Core/Def/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.cs

```diff
@@ -37,13 +37,13 @@ internal sealed partial class VisualStudioMetadataReferenceManager : IWorkspaceS
     private static readonly ConditionalWeakTable<Metadata, object> s_lifetimeMap = new();
 
     /// <summary>
-    /// Mapping from an <see cref="AssemblyMetadata"/> we created, to the memory mapped files (mmf) corresponding to
-    /// the assembly and all the modules within it.  This is kept around to make OOP syncing more efficient.
-    /// Specifically, since we know we read the assembly into an mmf, we can just send the mmf name/offset/length to
-    /// the remote process, and it can map that same memory in directly, instead of needing the host to send the
-    /// entire contents of the assembly over the channel to the OOP process.
+    /// Mapping from an <see cref="AssemblyMetadata"/> we created, to the identifiers identifying the memory mapped
+    /// files (mmf) corresponding to that assembly and all the modules within it.  This is kept around to make OOP
+    /// syncing more efficient. Specifically, since we know we dumped the assembly into an mmf, we can just send the mmf
+    /// name/offset/length to the remote process, and it can map that same memory in directly, instead of needing the
+    /// host to send the entire contents of the assembly over the channel to the OOP process.
     /// </summary>
-    private static readonly ConditionalWeakTable<AssemblyMetadata, IReadOnlyList<TemporaryStorageService.TemporaryStreamStorage>> s_metadataToStorages = new();
+    private static readonly ConditionalWeakTable<AssemblyMetadata, IReadOnlyList<TemporaryStorageIdentifier>> s_metadataToStorageIdentifiers = new();
```

Comment:
the prior patterns were much more confusing.  We'd be holding onto things like disposable object, with no commetns about lifetime management, and no actual lifetime management.  Things chnage here so that we only point onto the non-disposable objects representing what we need.